core/src/main/java/io/atomix/core/map/impl/AbstractAtomicMapService.java

```diff
@@ -0,0 +1,1058 @@
+/*
+ * Copyright 2016-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.core.map.impl;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import io.atomix.core.iterator.impl.IteratorBatch;
+import io.atomix.core.map.AtomicMapEvent;
+import io.atomix.core.transaction.TransactionId;
+import io.atomix.core.transaction.TransactionLog;
+import io.atomix.core.transaction.impl.CommitResult;
+import io.atomix.core.transaction.impl.PrepareResult;
+import io.atomix.core.transaction.impl.RollbackResult;
+import io.atomix.primitive.PrimitiveType;
+import io.atomix.primitive.service.AbstractPrimitiveService;
+import io.atomix.primitive.service.BackupInput;
+import io.atomix.primitive.service.BackupOutput;
+import io.atomix.primitive.session.Session;
+import io.atomix.primitive.session.SessionId;
+import io.atomix.utils.concurrent.Scheduled;
+import io.atomix.utils.serializer.Namespace;
+import io.atomix.utils.serializer.Serializer;
+import io.atomix.utils.time.Versioned;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * State Machine for {@link AtomicMapProxy} resource.
+ */
+public abstract class AbstractAtomicMapService<K> extends AbstractPrimitiveService<AtomicMapClient> implements AtomicMapService<K> {
+
+  private static final int MAX_ITERATOR_BATCH_SIZE = 1024 * 32;
```

Comment:
batch size too large?

Reply:
Ahh yeah, I think this needs to be changed. It was in bytes before, but it's not in entries IIRC.

Reply:
Nvm no it isn't.